### PR TITLE
Setup nx workspace

### PR DESCRIPTION
**Pull request by unicop**
# motivation

Convert into mono-repo

### CODE REVIEW

**Comment by unicop** [comment]
@neri-snappy 💥 
I adjust the `circleci` to work with nx!

File: .vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "recommendations": [
+    "nrwl.angular-console",
```
Comment (neriyarden) [inline .vscode/extensions.json:3]: angular?

File: .vscode/extensions.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+    "esbenp.prettier-vscode",
```
Comment (neriyarden) [inline .vscode/extensions.json:4]: also prettier is unnecessary if you have eslint extension

File: .husky/pre-commit [hunk 3]
```diff
@@ -1,4 +1,4 @@
 #!/bin/sh
 . "$(dirname "$0")/_/husky.sh"
 
-npx lint-staged
\ No newline at end of file
+# npx lint-staged
```
Comment (neriyarden) [inline .husky/pre-commit:4]: why is it commented?

File: .vscode/extensions.json (hunk 2 above)
Comment (unicop) [reply .vscode/extensions.json:4]: you're right ill remove both and recommend to download eslint and typescript exts

File: .husky/pre-commit (hunk 3 above)
Comment (unicop) [reply .husky/pre-commit:4]: I wanted to check that reviewdog is working, ill uncomment it and remove the eslint issue

File: .vscode/extensions.json (hunk 1 above)
Comment (unicop) [reply .vscode/extensions.json:3]: nrwl is the company behind nx, and this is how they've called to their vs extension, so it is not about angular but about nx :)

File: README.md [hunk 4]
```diff
@@ -1,46 +1,94 @@
-<h1 align="center">Migrate To Typescript</h1>
```
[the remark this answers is not in the document]
Comment (unicop) [reply README.md:1]: we should check what nx is writing there, maybe we can use part of it.
remember that we are about to have docs as well so the readme will remain unknown for some time until we will have more clear vision what it should present.

**Pull request merged**